docs/grpc/c#.md

```diff
@@ -0,0 +1,219 @@
+```
+
+* Copy Google's [annotations.proto](https://github.com/googleapis/googleapis/blob/master/google/api/annotations.proto) to the correct folder:
+```bash
+mkdir google
+mkdir google/api
+curl -o google/api/annotations.proto -s https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/annotations.proto
+```
+
+* Copy Google's [http.proto](https://github.com/googleapis/googleapis/blob/master/google/api/http.proto) to the correct folder:
+```bash
+curl -o google/api/http.proto -s https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/http.proto
+```
+
+* Copy Google's [descriptor.proto](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto) to the correct folder:
+```bash
+mkdir google/protobuf
+curl -o google/protobuf/descriptor.proto -s https://raw.githubusercontent.com/protocolbuffers/protobuf/master/src/google/protobuf/descriptor.proto
+```
+
+* Compile the proto file:
+```bash
+./protoc.exe --csharp_out . --grpc_out . rpc.proto --plugin=protoc-gen-grpc=grpc_csharp_plugin.exe
+```
+
+After following these steps, two files `Rpc.cs` and `RpcGrpc.cs` will be generated in the current directory. These files will need to be imported in your project any time you use C# `gRPC`.
+
+
+
+#### Imports and Client
+
+Every time you use C# `gRPC`, you will have to import the generated rpc classes, and use `nuget` package manger to install `Grpc.Core`, `Google.Protobuf`, and `Google.Api.CommonProtos`.
+
+After installing these, use the code below to set up a channel and client to connect to your `lnd` node:
+
+```c#
+
+using Grpc.Core;
+using Lnrpc;
+...
+
+// Due to updated ECDSA generated tls.cert we need to let gprc know that
+// we need to use that cipher suite otherwise there will be a handshake
+// error when we communicate with the lnd rpc server.
+System.Environment.SetEnvironmentVariable("GRPC_SSL_CIPHER_SUITES", "HIGH+ECDSA");
+            
+// Lnd cert is at AppData/Local/Lnd/tls.cert on Windows
+// ~/.lnd/tls.cert on Linux and ~/Library/Application Support/Lnd/tls.cert on Mac
+var cert = File.ReadAllText(<Tls_Cert_Location>);
+
+var sslCreds = new SslCredentials(cert);
+var channel = new Grpc.Core.Channel("localhost:10009", sslCreds);
+var client = new Lnrpc.Lightning.LightningClient(channel);
+
+```
+
+### Examples
+
+Let's walk through some examples of C# `gRPC` clients. These examples assume that you have at least two `lnd` nodes running, the RPC location of one of which is at the default `localhost:10009`, with an open channel between the two nodes.
+
+#### Simple RPC
+
+```c#
+// Retrieve and display the wallet balance
+// Use "WalletBalanceAsync" if in async context
+var response = client.WalletBalance(new WalletBalanceRequest());
+Console.WriteLine(response);
+```
+
+#### Response-streaming RPC
+
+```c#
+var request = new InvoiceSubscription();
+using (var call = client.SubscribeInvoices(request))
+{
+    while (await call.ResponseStream.MoveNext())
```

Comment:
Can definitely do that, but I thought it better to always get the latest versions as this project progresses so quickly! Better to have docs that are 95% correct than ones that are quickly out of date, in my opinion. Your thoughts?

Reply:
I don't think the `protoc` version is moving that quickly. I think for example code, it doesn't hurt to add enough details so that a user can run it. Maybe accompanied by a note that says that actual lnd may already be on a higher version?

Reply:
can't reproduce on windows using .net core. `MoveNext()` has a signature with and without a cancellation token. Maybe the async context caused the issue

Reply:
Can't reproduce it anymore myself. Probably we had different version previously and not anymore now.